### PR TITLE
Revised non Suppressed register

### PR DESCRIPTION
**Pull request by slubwama**
This is a recreated because of so many commits that couldn't be rebased Previous #350 

Ticket https://trello.com/c/Bo3R6bba/661-add-changes-of-the-revised-viral-load-non-suppressed-register

### CODE REVIEW

File: omod/src/main/webapp/resources/js/aijar.js [hunk 1]
```diff
@@ -161,25 +167,33 @@ function validateRequiredField(prime, factor, message_to_throw, input_type) {
     var selected_value = null;
     getField(prime + '.error').html("").hide;
 
-    if (input_type == "select") {
-        selected_value = jq(factor).find(":selected").text().trim().toLowerCase();
-        if (selected_value != '' && getValue(prime + '.value') == '') {
+    if (input_type === "select") {
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/js/aijar.js]: @slubwama Can you verify that these changes to not affect the ART Summary page where these functions are used for validation

File: omod/src/main/webapp/resources/js/aijar.js [hunk 2]
```diff
@@ -35,6 +35,12 @@ jq(document).ready(function () {
             phone_number.match(/^[0-9]{1,10}$/);
     }, "Please specify a valid mobile number without any spaces like 0712345678");
 
+    jq.validator.addMethod("nationalid", function(nationalid, element) {
+        nationalid = nationalid.replace(/\(|\)|\s+|-/g, "");
+        return this.optional(element) || nationalid.match(/^$|^[A-Z][FM]\d{5}([A-Z0-9]){7}$/);
+    }, "Enter a valid National ID example XX11112222XYX5");
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/js/aijar.js]: @slubwama National ID Example: use CF12345678ABCD (the duplicated numbers are hard to make sense of)

**Comment by ssmusoke** [comment]
@slubwama 

1. Date Results given to client cannot be after Date of Registration 
2. Regimen Start date is currently validating as must be after registration date, yet should be before.
3. Change "Start Date" to "Regimen Start Date" to be clear to the user  
3. Serum Crag screening result should only be entered when the rest is done 
4. Change IAC Session Date error message to read "Must be on or after date of Registration" not start date 
5. IAC Session start date validates against Regimen Start date instead of Date of Registration 
6. VL after IAC - when you select Not Detected a value of 1 should be put in copies per ml

**Comment by slubwama** [comment]
@ssmusoke  done responding to comments

**Comment by ssmusoke** [comment]
@slubwama 

1. Date Results given to client cannot be before date results received at the health facility 

2. Cannot submit with the values in the attachment below not sure what is not valid since an error pops up but no message is displayed

screenshot 2018-10-01 19 12 30

**Comment by ssmusoke** [comment]
@slubwama Please put more detailed commit messages to describe what has changed other than "Response to comments"

**Comment by slubwama** [comment]
> @slubwama Please put more detailed commit messages to describe what has changed other than "Response to comments"

Cant recommit with better message now it too late. Just review what I have commited

**Comment by ssmusoke** [comment]
@slubwama

1. Restart IAC, Adherence Codes, HIVDR sample collected - make a radio button since it only have 2 values 
2. IAC Session - when you save more than one session, they are not all displayed on editing as the additional rows are hidden

**Comment by slubwama** [comment]
@ssmusoke I have response to comments

**Comment by slubwama** [comment]
@ssmusoke waiting for your reviews

**Comment by ssmusoke** [comment]
@slubwama I cannot add IAC sessions when saving the form the first time see image below 
screen shot 2018-10-16 at 13 51 49

**Comment by slubwama** [comment]
@ssmusoke  thats done

**Pull request merged**